### DISCUSSION TITLE
Preprocessing for images

### DISCUSSION THREAD

**Discussion by sebffischer** [mlr-org/mlr3torch#18] [Ideas]
I have an idea how to implement the preprocessing for images. 
I general I think we want something like
```r
graph = top("resize", size = c(128, 128)) %>>% 
  po("learner", lrn("classif.alexnet"))
glrn = as_learner(graph)
```

In the train method of the alexnet learner, we somewhere do something like: 

```r
dl = as_dataloader(task)
```
I think the `top("resize", size = c(128,128))` should essentially 
take the `uri_column` in the task, and do something like: 

```r
attr(uri_column, "trafo") = c(attr(uri_column, "trafo"), function(x) transform_resize(x, size = c(128, 128))
```

The dataloader should then simply take these attributes (don't really have to be attributes but something assigned to the column) into account when calling their `$.getitem` method. This way we can pretend that the transformation is immediately applied to the images (making it conceptually similar to other preprocessing such as PCA for tabular data), because as soon as the `as_dataloader()` function is called, the transformations are applied.

**Comment by jemus42** [comment]
Would be cool if something like that worked 🤔 
Would multiple transformation `top`s then "stack" to create a sequential transformation?

There's at least one transformation that has to be applied every time: [`transform_to_tensor`](https://torchvision.mlverse.org/reference/transform_to_tensor.html), but since that would come before any transformation anyway that might as well be a hard-coded first step, along with loading the image from disk via `magick` (or whichever loader we end up using). 

Directly after that transformation we'll have to deal with device placement, i.e. moving the image tensor to a GPU (which involves detecting if one is available, or choosing one if multiple are available).
